clients/csrf/csrf_token_fetcher.go

```diff
@@ -0,0 +1,7 @@
+package csrf
+
+import "net/http"
+
+type CsrfTokenFetcher interface {
+	FetchNewCsrfToken(url string, currentRequest *http.Request) (string, string, error)
```

Comment:
Personally, I do not like more than 3 args if they needed indeed. Could a return object be introduced here and return it directly?

Reply:
Fixed, a new object is returned